### ISSUE TITLE
Database and SQL related properties/data libraries

### ISSUE THREAD

**Issue by Thelulucien** [cerberustesting/cerberus-source#1832]
Hi,

I'm currently working on several tests that need to check a specific value in a table and to compare it.

This table is in a remote database and this DB is not proposed in Cerberus.
The fact that it is not proposed was expected but I didn't find how to add it.

I saw nothing in the doc related to this issue, I just saw where to write my sql script when the DB is proposed and even there I'm a bit lost : I started Cerberus yesterday !

Also, I didn't understand the difference between this : 
![data_library_db_pb](https://user-images.githubusercontent.com/17575025/45104306-1a265100-b132-11e8-9071-20e1798dcbeb.png)

And this : 
![property_db_pb](https://user-images.githubusercontent.com/17575025/45104353-36c28900-b132-11e8-979c-539bfe16452c.png)

I know that data library have a larger scope than properties but it seems to be more than that.

Short version : I don't know how to ask cerberus to add my DB and to get data using SQL.

Thank you and excuse my English : I'm a baguette eater.

**Comment by vertigo17** [comment]
DB can be added inside invariant menu:
![image](https://user-images.githubusercontent.com/5342048/45116543-55854780-b153-11e8-9ee8-61aa10396fa1.png)
That will allow to add the entry on both database field of your screenshot.
NB : You may need to login logout in order to have it appear in combos.

Once added you will need to configure the database on the corresponding environment in order to point to the correct jdbc ressource.
![image](https://user-images.githubusercontent.com/5342048/45116833-36d38080-b154-11e8-98b9-760ef1dcb2f9.png)

Last step is to create from glasfish administration console the jdbc/name ressource (this is where the ip/port/databasename/user/pass needs to be configured). You also need to put the correct jdbc drivers on lib folder of glassfish.

Then about difference between getFromSql and getFromDataLib is:
getFromSql : 
pro: easier and quicker to define 
cons: SQL is hard code in every test. in case of change you need to change all test properties
cons: You can only get a single string result (1st colum of 1st row).
With getFromDatalib:
pro: all SQL definition is centralise and can be reused from 1 testcase to another
pro: A lot more powerfull as you get an object (multi row and multi column)
pro: You can change your mind switch from csv source or REST call without any impact on testcase definition.
cons: more complex to define and administer and require mapping between sql result and subdata names.

My advice is to start with getFromSql and start using getFromDataLib in a second step.

**Comment by Thelulucien** [comment]
Re !

I did discuss this with my superior and he said that the tests I was working on wasn't the priority for the moment so I didn't not have the opportunity to try this : I don't have the authorization to manipulate drivers on the system.

Anyway, in a few weeks we will have to implement it so I will come back here to give a feedback on the issue.

Thanks for the quick and complete answer and thanks again for your tool !

Bonne journée !
